pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -0,0 +1,763 @@
+	// It is the name that plugins use in their pod mount path, i.e.
+	// /var/lib/kubelet/pods/{podUID}/volumes/{escapeQualifiedPluginName}/{innerVolumeSpecName}/
+	// PVC example,
+	//   apiVersion: v1
+	//   kind: PersistentVolume
+	//   metadata:
+	//     name: pv0003				<- InnerVolumeSpecName
+	//   spec:
+	//     capacity:
+	//       storage: 5Gi
+	//     accessModes:
+	//       - ReadWriteOnce
+	//     persistentVolumeReclaimPolicy: Recycle
+	//     nfs:
+	//       path: /tmp
+	//       server: 172.17.0.2
+	// Non-PVC example:
+	//   apiVersion: v1
+	//   kind: Pod
+	//   metadata:
+	//     name: test-pd
+	//   spec:
+	//     containers:
+	//     - image: gcr.io/google_containers/test-webserver
+	//     	 name: test-container
+	//     	 volumeMounts:
+	//     	 - mountPath: /test-pd
+	//     	   name: test-volume
+	//     volumes:
+	//     - name: test-volume			<- InnerVolumeSpecName
+	//     	 gcePersistentDisk:
+	//     	   pdName: my-data-disk
+	//     	   fsType: ext4
+	InnerVolumeSpecName string
+
+	// outerVolumeSpecName is the podSpec.Volume[x].Name of the volume. If the
+	// volume was referenced through a persistent volume claim, this contains
+	// the podSpec.Volume[x].Name of the persistent volume claim.
+	// PVC example:
+	//   kind: Pod
+	//   apiVersion: v1
+	//   metadata:
+	//     name: mypod
+	//   spec:
+	//     containers:
+	//       - name: myfrontend
+	//         image: dockerfile/nginx
+	//         volumeMounts:
+	//         - mountPath: "/var/www/html"
+	//           name: mypd
+	//     volumes:
+	//       - name: mypd				<- OuterVolumeSpecName
+	//         persistentVolumeClaim:
+	//           claimName: myclaim
+	// Non-PVC example:
+	//   apiVersion: v1
+	//   kind: Pod
+	//   metadata:
+	//     name: test-pd
+	//   spec:
+	//     containers:
+	//     - image: gcr.io/google_containers/test-webserver
+	//     	 name: test-container
+	//     	 volumeMounts:
+	//     	 - mountPath: /test-pd
+	//     	   name: test-volume
+	//     volumes:
+	//     - name: test-volume			<- OuterVolumeSpecName
+	//     	 gcePersistentDisk:
+	//     	   pdName: my-data-disk
+	//     	   fsType: ext4
+	OuterVolumeSpecName string
+
+	// PluginName is the "Unescaped Qualified" name of the volume plugin used to
+	// mount and unmount this volume. It can be used to fetch the volume plugin
+	// to unmount with, on demand. It is also the name that plugins use, though
+	// escaped, in their pod mount path, i.e.
+	// /var/lib/kubelet/pods/{podUID}/volumes/{escapeQualifiedPluginName}/{outerVolumeSpecName}/
+	PluginName string
+
+	// PodUID is the UID of the pod mounted to. It is also the string used by
+	// plugins in their pod mount path, i.e.
+	// /var/lib/kubelet/pods/{podUID}/volumes/{escapeQualifiedPluginName}/{outerVolumeSpecName}/
+	PodUID types.UID
+
+	// Mounter is the volume mounter used to mount this volume. It is required
+	// by kubelet to create container.VolumeMap.
+	Mounter volume.Mounter
+}
+
+type operationExecutor struct {
```

Comment:
Nope. `volumePluginMgr` is pre-initialized and is only used to get plugins (i.e. reads only). And pendingOperations is already thread-safe via internal mutex.